sqlglot/tokens.py

```diff
@@ -1033,15 +1033,19 @@ def _scan_bits(self) -> None:
         self._advance()
         value = self._extract_value()
         try:
-            self._add(TokenType.BIT_STRING, f"{int(value, 2)}")
+            # If `value` can't be converted to a binary, fallback to tokenizing it as an identifier
+            int(value, 2)
+            self._add(TokenType.BIT_STRING, value[2:])  # Drop the 0b
         except ValueError:
             self._add(TokenType.IDENTIFIER)
 
     def _scan_hex(self) -> None:
         self._advance()
         value = self._extract_value()
         try:
-            self._add(TokenType.HEX_STRING, f"{int(value, 16)}")
+            # If `value` can't be converted to a hex, fallback to tokenizing it as an identifier
+            int(value, 16)
+            self._add(TokenType.HEX_STRING, value[2:])  # Drop the 0x
```

Comment:
So this is the crucial bit, right?

Reply:
There were a couple other parts that needed to be changed as well. Minor changes though, yeah.